core-web/libs/sdk/client/src/lib/editor/listeners/listeners.ts

```diff
@@ -45,29 +39,43 @@ function setBounds(): void {
  * @memberof DotCMSPageEditor
  */
 export function listenEditorMessages(): void {
-    const messageCallback = (event: MessageEvent) => {
-        switch (event.data) {
-            case NOTIFY_CUSTOMER.EMA_REQUEST_BOUNDS: {
+    const messageCallback = (
+        event: MessageEvent<{ name: NOTIFY_CLIENT; direction: 'up' | 'down' }>
+    ) => {
+        const ACTIONS_NOTIFICATION: Record<NOTIFY_CLIENT, () => void> = {
+            [NOTIFY_CLIENT.UVE_RELOAD_PAGE]: () => {
+                window.location.reload();
+            },
+            [NOTIFY_CLIENT.UVE_REQUEST_BOUNDS]: () => {
                 setBounds();
-                break;
-            }
-        }
-
-        if (event.data.name === NOTIFY_CUSTOMER.EMA_SCROLL_INSIDE_IFRAME) {
-            const direction = event.data.direction;
-
-            if (
-                (window.scrollY === 0 && direction === 'up') ||
-                (scrollIsInBottom() && direction === 'down')
-            ) {
-                // If the iframe scroll is at the top or bottom, do not send anything.
-                // This avoids losing the scrollend event.
-                return;
+            },
+            [NOTIFY_CLIENT.UVE_SCROLL_INSIDE_IFRAME]: () => {
+                const direction = event.data.direction;
+
+                if (
+                    (window.scrollY === 0 && direction === 'up') ||
+                    (scrollIsInBottom() && direction === 'down')
+                ) {
+                    // If the iframe scroll is at the top or bottom, do not send anything.
+                    // This avoids losing the scrollend event.
+                    return;
+                }
+
+                const scrollY = direction === 'up' ? -120 : 120;
+                window.scrollBy({ left: 0, top: scrollY, behavior: 'smooth' });
+            },
+            [NOTIFY_CLIENT.UVE_SET_PAGE_DATA]: () => {
+                /** NOOP **/
+            },
+            [NOTIFY_CLIENT.UVE_COPY_CONTENTLET_INLINE_EDITING_SUCCESS]: () => {
+                /** NOOP **/
+            },
+            [NOTIFY_CLIENT.UVE_EDITOR_PONG]: () => {
+                /** NOOP **/
             }
```

Comment:
What? why?

Reply:
Didn't want to do type gimnastics when I wrote this. I fixed it now